src/main/java/org/spongepowered/api/event/entity/MoveEntityEvent.java

```diff
@@ -27,36 +27,38 @@
 import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.event.Cancellable;
 import org.spongepowered.api.event.Event;
+import org.spongepowered.api.util.annotation.eventgen.AbsoluteSortPosition;
 import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;
 import org.spongepowered.api.world.server.ServerWorld;
-import org.spongepowered.api.world.teleport.PortalAgent;
 import org.spongepowered.math.vector.Vector3d;
 
 /**
  * Called when an {@link Entity} performs movement.
  */
-@GenerateFactoryMethod
 public interface MoveEntityEvent extends Event, Cancellable {
```

Comment:
I think this should contain `getServerWorld()` or `getServerLocation()` as many plugins will be interested in the entity's location, which includes the world they are in, and will want to act upon that. Don't need a setter, but a getter will support plugin devs better.